WalletWasabi.Fluent/ViewModels/Wallets/Home/History/HistoryViewModel.cs

```diff
@@ -105,7 +105,15 @@ private async Task UpdateAsync()
 				{
 					if (_transactions.FirstOrDefault(x => x.Id == newItem.Id) is { } item)
 					{
-						item.Update(newItem);
+						if (item.GetType() != newItem.GetType())
```

Comment:
So from now if a transaction was detected as a CJ then later as a transaction (or vice versa) that is okay and we handle it as it would be a normal case?

Reply:
I think a few concepts here are problematic, but properly updating the UI based on the change of the underlying data structure is more consistent than throwing an exception.

Reply:
In the past, I thought it will never happen that the history table will try to update a txn item with a CJ item. But it happened and turned out there is a bug. It got fixed and later it came out again. Thanks to that exception both times we were aware of there is a bug somewhere. Now, it will not come out.